website/docs/tutorial/refactoring-legacy-sql.md

```diff
@@ -0,0 +1,249 @@
+
+Once you've copied it over, you'll want to `dbt run` to execute the query and populate the table in your warehouse.
+
+> If this is your first time running dbt, you may want to start with the [Introduction to dbt](/docs/introduction) and the [Getting Started tutorial](/tutorial/setting-up) before diving into refactoring.
+
+This step may sound simple, but if you're porting over an existing set of SQL transformations to a new SQL dialect, you will need to consider how your legacy SQL dialect differs from your new SQL flavor, and you may need to modify your legacy code to get it to run at all.  
+
+This will commonly happen if you're migrating from a [stored procedure workflow on a legacy database](https://getdbt.com/analytics-engineering/case-for-elt-workflow/) into dbt + a cloud data warehouse.
+
+Functions that you were using previously may not exist, or their syntax may shift slightly between SQL dialects. 
+
+If you're not migrating data warehouses at the moment, then you can keep your SQL syntax the same. You have access to the exact same SQL dialect inside of dbt that you have querying directly from your warehouse.
+
+## Create sources from table references
+
+<WistiaVideo id="m1a5p32rny" />
+
+To query from your data warehouse, we recommend creating [sources in dbt](/docs/building-a-dbt-project/using-sources) rather than querying the database table directly.
+
+This allows you to call the same table in multiple places with `{{ src('my_source', 'my_table') }}` rather than `my_database.my_schema.my_table`.
+
+We start here for several reasons:
+
+#### Source freshness reporting
+Using sources unlocks the ability to run [source freshness reporting](/docs/building-a-dbt-project/using-sources#snapshotting-source-data-freshness) to make sure your raw data isn't stale.
+
+#### Easy dependency tracing
+If you're migrating multiple stored procedures into dbt, with sources you can see which queries depend on the same raw tables. 
+
+This allows you to consolidate modeling work on those base tables, rather than calling them separately in multiple places. 
+
+<Lightbox src="/img/docs/building-a-dbt-project/sources-dag.png" title="Sources appear in green in your DAG in dbt docs" />
+
+#### Build the habit of analytics-as-code
+Sources are an easy way to get your feet wet using config files to define aspects of your transformation pipeline.
+
+```yml
+sources:
+  - name: jaffle_shop
+    tables:
+      - name: orders
+      - name: customers
+```
+
+With a few lines of code in a `.yml` file in your dbt project's `/models` subfolder, you can now version control how you your data sources (Snowplow, Shopify, etc) map to actual database tables.
+
+For example, say you migrate from one [ETL tool](https://getdbt.com/analytics-engineering/etl-tools-a-love-letter/) to another, and the schema name on your tables changes from - source config files allow you to make that migration from a single file, and flip the change on with one pull request to your repo. 
```

Comment:
```suggestion
For example, say you migrate from one [ETL tool](https://getdbt.com/analytics-engineering/etl-tools-a-love-letter/) to another, and the schema name on your tables changes from - source config files allow you to make that migration from a single file, and flip the change on with one pull request to your repo. 
```
your tables changes from....you were on to something but I don't think it made its way in.